CRM/Financial/BAO/FinancialAccount.php

```diff
@@ -246,4 +246,43 @@ public static function getFinancialAccountForFinancialTypeByRelationship($financ
     return Civi::$statics[__CLASS__]['entity_financial_account'][$financialTypeID][$relationTypeId];
   }
 
+  /**
+   * Get Financial Account type relations
+   *
+   * @param $flip bool
```

Comment:
Why did you decide to set $flip? Is this a pattern copied from somewhere else? 

I think it's common but bad pattern in CiviCRM - it adds code & complexity when the calling function could just use foreach(array_flip($result)) 

I also hate the idea of having $id as a filter parameter - I much prefer to see 2 functions

1) getfinancialAccountRelations
2) getfinancialAccountRelationsByID 

The latter can call the former.